### DISCUSSION TITLE
IC-Light V2-Vary

### DISCUSSION THREAD

**Discussion by lllyasviel** [lllyasviel/IC-Light#109] [Announcements]
Alternative IC-Light V2 Model(s) for people who want **stronger illumination variations and modifications**.

### Demo

here: https://huggingface.co/spaces/lllyasviel/iclight-v2-vary

### Examples

Input image:

<img src="https://github.com/user-attachments/assets/797a1713-5b4e-44a3-aa2b-e8fe6a9bdb1a" width="256">

Outputs Comparision (to see the capability to change light and shadow):

![gifmaker_me](https://github.com/user-attachments/assets/b5bcf73f-dbc5-4b16-aa71-3f0ef08966e2)

Input-output Comparison (to see the detail preservation level):

![gifmaker_me (2)](https://github.com/user-attachments/assets/936bd8e4-5603-4016-9ab4-951270f7f3c3)

Screenshots:

![image](https://github.com/user-attachments/assets/9a676b90-ea66-43f8-8808-f5f60aa3a8ad)

---

![gifmaker_me (3)](https://github.com/user-attachments/assets/57f845d0-fa46-481a-9e45-324767653c8b)

![image](https://github.com/user-attachments/assets/97cb51b3-5eaf-43f9-9a36-906873a4ffa3)

![image](https://github.com/user-attachments/assets/308298bf-0be7-494a-ac8d-1518676aff24)

![gifmaker_me (4)](https://github.com/user-attachments/assets/200d7dff-5f71-401a-9c1d-f70bf025eedc)

![image](https://github.com/user-attachments/assets/0aaf651f-55b3-49af-aac8-df2a4a7738a9)

![image](https://github.com/user-attachments/assets/0b72b630-0f17-4311-9ed3-71da99113556)

---

![gifmaker_me (1)](https://github.com/user-attachments/assets/38cd7086-7156-41ed-8a15-dd4ee496c269)

![image](https://github.com/user-attachments/assets/ae5ada9b-1f06-4105-b321-d113bcf17082)

![image](https://github.com/user-attachments/assets/0d7be5e4-93c2-4412-ad0f-1d92330428f5)

![gifmaker_me (5)](https://github.com/user-attachments/assets/d8a7a7ef-f2c5-4ee9-9d6b-f485f912d807)

---

![gifmaker_me (6)](https://github.com/user-attachments/assets/2f537c7b-2ec5-4693-b475-864e44577d83)

![image](https://github.com/user-attachments/assets/bbb037f3-32c5-433f-9e4f-b6c1074b4a04)

---

![image](https://github.com/user-attachments/assets/1f800f41-db85-4609-95d8-05ce9e499da1)

![image](https://github.com/user-attachments/assets/a3cbd075-dde0-4aab-b1c5-d524b9d569f6)

![image](https://github.com/user-attachments/assets/e5885fa6-39e0-4ef9-9a7b-2b9d430ff086)

Note that all examples above can be reproduced by clicking examples in the HuggingFace demo at the beginning of this post.

**Comment by bsdesigner** [comment]
That's great! Thank you so much for your work!
![Снимок экрана 2024-11-28 083212](https://github.com/user-attachments/assets/17df8f7c-167f-4c0d-aba1-afd5a4b69449)

**Reply by elizajpeg** [reply]
How can I use this tool? I'm new to github and I'm a professional portrait photographer. Do help me with it.

**Comment by zombieyang** [comment]
will it opensource?

**Reply by chigozie316** [reply]
Uploading v09044g40000ctsq1bnog65otn2tbo9g.mp4…

**Comment by T8mars** [comment]
looking forward to your opensource!!

**Comment by coolmian** [comment]
Thanks for your work!

**Comment by ericxl277** [comment]
thanks for you work,looking forward to your opensource!

**Comment by SunGreen777** [comment]
Will this be added to forge (space)? Are you planning to update something with sd 3.5 support and other new features ?

**Comment by MidnightFiend** [comment]
after relighting the portrait photo with demo, there is a serious color difference between the color of the subject's eyelids and that of the face.

**Reply by MidnightFiend** [reply]
After relighting, the skin may look sallow.

**Comment by 13426447442** [comment]
looking forward to your opensource!!

**Comment by buzhidao123abc** [comment]
looking forward to your opensource!!

**Comment by Dazzastrous** [comment]
Amazing work you do  lllyasviel.

**Comment by Nerual-Dreming** [comment]
@lllyasviel Hello, once again you have created an excellent technology. Are you planning to open its code? I noticed that you seem to have hidden your code in hf sapce for the first time, and it still has not been published. Perhaps there are some reasons for this. Should we expect the code at all?

**Comment by ok52000** [comment]
lllyasviel666

**Comment by g29times** [comment]
great work

**Comment by GENKAIx** [comment]
Nice

**Comment by dinelkathilina** [comment]
Wow, it's superb! I hope it will remain open-source.

**Comment by liumeice** [comment]
🥺求求你，尽快开源吧！拜托了

**Comment by FurkanGozukara** [comment]
Still waiting V2 :(

Who wanna try v1 i shared installers : https://github.com/lllyasviel/IC-Light/issues/122

**Comment by LCLAI123** [comment]
谢谢，收到！

**Comment by BingHanOfUESTC** [comment]
COOL!!!!

**Comment by apo818** [comment]
都说了不开源了，还等啥，有很多都开始商用了

**Reply by seafoodtest** [reply]
哪裡可以看到作者說不開源了

**Comment by blackbrand662** [comment]
Guys how can i use unlimited? Author answer please i am a new user

**Comment by amrpegasus** [comment]
amazing

**Comment by malvinmaro** [comment]
Wow

**Comment by mikebilly** [comment]
when will the code be published?

**Comment by Maelstrom2014** [comment]
any news from best of the best?

**Comment by fairstonexxx** [comment]
Such an incredible model, looking forward for opensource

**Comment by LCLAI123** [comment]
谢谢，收到！

**Comment by SL126383** [comment]
How do I use this tools

**Reply by nikolaiusa** [reply]
Очень ждем веса брат)

**Comment by zombieyang** [comment]
Could it become a API node of comfyUI? It is worth to be paid.

**Comment by Milor123** [comment]
where can download it?

**Comment by nikolaiusa** [comment]
If there are no weights, how to do it yourself?) Do I need to train lora with low rank dimension and turn on only certain layers?

**Comment by Tchomian** [comment]
Comme modifier une photo

**Comment by lazarodagoberto** [comment]
como puedo instalar esto en sd FUSION instante instalar pero no me permite

**Reply by khansabir009024-code** [reply]
![IMG-20250821-WA0016](https://github.com/user-attachments/assets/802f84a8-ffad-422d-a89a-f380f7dc8bcf)
Isko Burj Khalifa ke pass banao

**Reply by khansabir009024-code** [reply]
Is me back ground laga ke do

**Reply by l33tx0** [reply]
use google Gemini nano banana https://gemini.google.com/app
